dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/Constants.java

```diff
@@ -27,7 +27,12 @@
  */
 package org.hisp.dhis.dxf2;
 
+import java.util.Arrays;
+import java.util.List;
+
 public final class Constants
 {
     public static final String UNKNOWN = "[Unknown]";
+
+    public static final List<String> PROGRAM_RULE_VARIABLE_NAME_INVALID_KEYWORDS = Arrays.asList( "and", "or", "not" );
```

Comment:
just for information. Will it also block AND|OR|NOT ( caps )?

Reply:
yes i had the same question in my mind but as described in the issue it should be only about lower case, can you confirm @enricocolasante ?

Reply:
Yes, the grammar is not accepting capital letter tokens, so there is no need to block them